discovery-provider/src/queries/get_user_listening_history.py

```diff
@@ -79,6 +79,13 @@ def _get_user_listening_history(session: Session, args: GetUserListeningHistoryA
     if not listening_history_results:
         return []
 
+    # order listening history entries by their user's play counts so our track ids will be
+    # correct order when querying for track ids
+    if sort_method == SortMethod.most_listens_by_user:
```

Comment:
confused by sorting here and then it sorts again in sort_by_sort_method?

Reply:
yea its not the best.. i had to sort track ids here using the play_count information available in user listening history. which then lets me query the tracks table for those sorted track ids in order (specified by the sort by sort method fn). ideally all the sorting would happen in sort_by_sort_method, but we don't have access to the play count info there, unless theres a way to access it there?

Reply:
ooh i see, thanks! this seems good